src/main/groovy/org/shipkit/internal/gradle/git/GitCheckOutTask.java

```diff
@@ -27,8 +36,45 @@ public String getRev() {
         return rev;
     }
 
+    /**
+     * See {@link #isNewBranch()}
+     */
+    public void setNewBranch(boolean newBranch) {
+        this.newBranch = newBranch;
+    }
+
+    /**
+     * Should checkout create a new branch
+     */
+    public boolean isNewBranch() {
+        return newBranch;
+    }
+
     @TaskAction
     public void checkOut() {
-        new DefaultProcessRunner(getProject().getProjectDir()).run("git", "checkout", rev);
+        getProcessRunner().run(getCommandLine());
+    }
+
+    private List<String> getCommandLine(){
+        List<String> commandLine = new ArrayList<String>();
+        commandLine.add("git");
+        commandLine.add("checkout");
+        if(newBranch){
+            commandLine.add("-b");
+        }
+        commandLine.add(rev);
+        return commandLine;
+    }
+
+    private ProcessRunner getProcessRunner(){
+        if(processRunner == null){
+            return new DefaultProcessRunner(getProject().getProjectDir());
+        }
+        return processRunner;
+    }
+
+    @ExposedForTesting
+    protected void setProcessRunner(ProcessRunner processRunner){
```

Comment:
I suggest to push out all implementation details out of GitCheckOutTask. See how it is done in ReleaseNeededTask. To me, it feels that  "@ExposedForTesting" is not really needed given that the type that holds the method is internal anyway. We can also keep methods exposed for testing as package-protected.

Good discussion!